src/storage/txn/store.rs

```diff
@@ -232,50 +222,46 @@ impl Store for FixtureStore {
             }
         });
 
-        let mut vec: Vec<_> = self
-            .data
-            .range((lower, upper))
-            .map(|(k, v)| {
-                let owned_k = k.clone();
-                let owned_v = if key_only {
-                    match v {
-                        Ok(_v) => Ok(vec![]),
-                        Err(e) => Err(e.maybe_clone().unwrap()),
-                    }
-                } else {
-                    match v {
-                        Ok(v) => Ok(v.clone()),
-                        Err(e) => Err(e.maybe_clone().unwrap()),
-                    }
-                };
-                (owned_k, owned_v)
-            })
-            .collect();
-
-        if desc {
-            vec.reverse();
-        }
-
         Ok(FixtureStoreScanner {
-            // TODO: Remove clone when GATs is available. See rust-lang/rfcs#1598.
-            data: vec.into_iter(),
+            key_only,
+            desc,
+            handle: OwningHandle::new_with_fn(self.data.clone(), |m| unsafe {
+                Box::new((*m).range((lower, upper)))
+            }),
         })
     }
 }
 
 /// A Scanner that scans on fixtures.
+// TODO: Replace it using GATs when available, to avoid `Arc` cost.
+#[allow(clippy::type_complexity)]
 pub struct FixtureStoreScanner {
-    data: std::vec::IntoIter<(Key, Result<Vec<u8>>)>,
+    key_only: bool,
+    desc: bool,
+    handle: OwningHandle<
+        Arc<BTreeMap<Key, Result<Vec<u8>>>>,
+        Box<btree_map::Range<'static, Key, Result<Vec<u8>>>>,
```

Comment:
What if using `Box<Iterator>`?

Reply:
I believe you need to write `Box<Iterator<Item=&'?? ..>>` instead of a simple `Box<Iterator>` and again `&'??` cannot be written out.

Reply:
Isn't the Item `Result<(Key, Vec<u8>)>`?

Reply:
The `BTreeMap`'s item itself is a reference (and we will clone to an owned value as our return value).

The point of this PR is that, we usually assign a very large range, let's say, `[-inf, +inf)`. However we will only consume some part of it. So it is not necessary to pre-clone everything in `[-inf, +inf)`.

Reply:
What I mean is boxing L235~L252 as an iterator, so it's still copied in need and get a simple structure.

Reply:
@BusyJay You need to write out the lifetime anyway I think. For example, if you wrap the lifetime in a structure that implements Iterator that outputs a `'static`, then what is the lifetime inside the field?

Reply:
No, you don't. The lifetime is 'static. And the default lifetime of Box is 'static. So precisely, the type you need to write is `OwningHandle<Arc<BTreeMap<Key, Result<Vec<u8>>>>, Box<dyn Iterator<Item = (Key, Result<Vec<u8>>)> + Send>>` and `key_only`, `desc` are unnecessary any more.